includes/admin/class-wc-stripe-inbox-notes.php

```diff
@@ -39,16 +39,18 @@ public static function notify_on_apple_pay_domain_verification( $verification_co
 			}
 
 			// If the domain verification completed after failure note was created, make sure it's marked as actioned.
-			$data_store       = WC_Data_Store::load( 'admin-note' );
-			$failure_note_ids = $data_store->get_notes_with_name( self::FAILURE_NOTE_NAME );
-			if ( ! empty( $failure_note_ids ) ) {
-				$note_id = array_pop( $failure_note_ids );
-				$note    = WC_Admin_Notes::get_note( $note_id );
-				if ( false !== $note && WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED !== $note->get_status() ) {
-					$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED );
-					$note->save();
+			try {
+				$data_store       = WC_Data_Store::load( 'admin-note' );
+				$failure_note_ids = $data_store->get_notes_with_name( self::FAILURE_NOTE_NAME );
+				if ( ! empty( $failure_note_ids ) ) {
+					$note_id = array_pop( $failure_note_ids );
+					$note    = WC_Admin_Notes::get_note( $note_id );
+					if ( false !== $note && WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED !== $note->get_status() ) {
+						$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED );
+						$note->save();
+					}
 				}
-			}
+			} catch ( Exception $e ) {}  // @codingStandardsIgnoreLine.
```

Comment:
Instead of a try/catch with an empty catch block, maybe it would be better to wrap with [`method_exists`](https://www.php.net/manual/en/function.method-exists.php)?

```php
if ( method_exists( 'WC_Data_Store', 'load' ) ) {
    // ...
}
```

Reply:
And maybe we should log an error indicating that the merchant might want to update WC to get the data store?

Reply:
Actually, we do need the try/catch, nevermind!

Didn't read the error message carefully enough 😬